### PR TITLE
server ownership verification

### PR DESCRIPTION
**Pull request by Southclaws**

### CODE REVIEW

File: server/src/api/servers/h_verify.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+//
+// { "code": "somerandomcode" } the code the user needs to set their mapname to
+//
+// subsequent responses:
+//
+// { "verified": true }      server ownership verified
+// { "verified": false }     server ownership not verified, aborting
+// { "error": "some error" } not verified, error occurred, retrying
+//
+func (s *service) vertify(w http.ResponseWriter, r *http.Request) {
```
Comment (Alasnkz) [inline server/src/api/servers/h_verify.go:29]: 🕯

**Review by Alasnkz — APPROVED** [review]

File: server/src/api/servers/h_verify.go (hunk 1 above)
Comment (Southclaws) [reply server/src/api/servers/h_verify.go:29]: 🙏

**Comment by Alasnkz** [comment]
Have you considered what to do if 2 people attempt to verify the same server? i.e.

Person A is a previous server-owner on some shoddy sa-mp slot server host, now Person B owns it but Person A verified it before, how would that work?

**Comment by Southclaws** [comment]
> 
> 
> Have you considered what to do if 2 people attempt to verify the same server? i.e.
> 
> Person A is a previous server-owner on some shoddy sa-mp slot server host, now Person B owns it but Person A verified it before, how would that work?
> 
> This just came to me now before my initial approval

Yeah this is a difficult situation, I think the only way to solve this is verification timing out and needing to be re-verified. For now, I think those situations probably happen rare enough to be handled case-by-case until a more permanent solution is put in place.

A permanent solution would probably be a custom `announce` binary and an API token associated with the user. Once the new server owner takes over, they would discard the old token and create their own token to run the announce binary with (either in server.cfg or some other config source).

But the issue with a custom announce binary is it's a lot more work to do that - especially for sampctl users (but sampctl can help automate that @ADRFranklin) so for now, the mapname solution is fairly simple - it's a one-time command run from server terminal/rcon/in game.

**Comment by BigETI** [comment]
> 
> 
> > Have you considered what to do if 2 people attempt to verify the same server? i.e.
> > Person A is a previous server-owner on some shoddy sa-mp slot server host, now Person B owns it but Person A verified it before, how would that work?
> > This just came to me now before my initial approval
> 
> Yeah this is a difficult situation, I think the only way to solve this is verification timing out and needing to be re-verified. For now, I think those situations probably happen rare enough to be handled case-by-case until a more permanent solution is put in place.
> 
> A permanent solution would probably be a custom `announce` binary and an API token associated with the user. Once the new server owner takes over, they would discard the old token and create their own token to run the announce binary with (either in server.cfg or some other config source).
> 
> But the issue with a custom announce binary is it's a lot more work to do that - especially for sampctl users (but sampctl can help automate that @ADRFranklin) so for now, the mapname solution is fairly simple - it's a one-time command run from server terminal/rcon/in game.

Maybe it is worth extending the existing server list API to support authentication tokens, and maybe https://github.com/BigETI/SAMPServerAnnouncer can be updated to support this feature.

**Pull request merged**